python/execute.py

```diff
@@ -306,27 +312,39 @@ def visit_target(self, query):
     def visit_join(self, query):
         raise "Not implemented"
 
+    @staticmethod
+    def _visit_driver_obj(it, key, driver):
+        with ExitStack() as stack:
+            for i in it:
+                id_i = i[":id"]
+                res = driver(key, id_i)
+                for k, v in res.items():
+                    stack.callback(i.__setitem__, k, v)
```

Comment:
General problem: iterating over a container while modifying itself is tricky. The ExitStack solution was suggested as a workaround.